File: NumberShapes/app/src/main/java/com/example/numbershapes/MainActivity.java
package com.example.numbershapes;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        Button but = (Button)findViewById(R.id.button);
        EditText ed = (EditText)findViewById(R.id.Numb);





       but.setOnClickListener(new View.OnClickListener() {
           @Override
           public void onClick(View view) {


               Number no = new Number();
               if (ed.getText().toString().isEmpty()) {
                   Toast.makeText(MainActivity.this, "Please enter a number", Toast.LENGTH_LONG).show();
               } else {


                   no.number = Integer.parseInt(ed.getText().toString());

                   ed.onEditorAction(EditorInfo.IME_ACTION_DONE);
                   String fm = ed.getText().toString();

                   String message;


                   if (no.isSquareNumber() && no.isTriangular()) {
                       message = fm + "Is both a square and triangular number";
                       Toast.makeText(MainActivity.this, message, Toast.LENGTH_LONG).show();
                   } else if (no.isSquareNumber() && !no.isTriangular()) {
                       message = fm + "Is a Squarenumber";
                       Toast.makeText(MainActivity.this, message, Toast.LENGTH_LONG).show();
                   } else if (!no.isSquareNumber() && no.isTriangular()) {
                       message = fm + "Is a Triangular Numer";
                       Toast.makeText(MainActivity.this, message, Toast.LENGTH_LONG).show();
                   } else {
                       message = fm + "Neither a square nor a triangular number";
                       Toast.makeText(MainActivity.this, message, Toast.LENGTH_LONG).show();
                   }

               }
           }
       });

    }

    public class Number
    {
        int number;

        public boolean isTriangular()
        {
            int x=1;
            int triangularNumber = 1;
            while(triangularNumber<number)
            {
                x++;
                triangularNumber += x;
            }
            if(triangularNumber==number)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public boolean isSquareNumber()
        {
            double sqroot=Math.sqrt(number);
            if(sqroot == Math.floor((sqroot)))
            {
                return true;
            }
            else
            {
                return false;
            }
        }

    }
}

File: Notes App/app/src/main/java/com/example/mynotes1/MainActivity.kt
package com.example.mynotes1

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.findNavController
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.mynotes1.Note.*
import com.google.android.material.floatingactionbutton.FloatingActionButton



class MainActivity : AppCompatActivity(),OnItemClickListener {

    private val newNoteActivityRequestCode = 1

    val EXTRA_MESSAGE = "com.example.myfirstapp.MESSAGE"

    private val noteViewModel: NoteViewModel by viewModels {
        NoteViewModelFactory((application as NotesApplication).repository)
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView = findViewById<RecyclerView>(R.id.recyclerview)
        val adapter = NoteListAdapter(this)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)

        noteViewModel.allNotes.observe(this, Observer { notes ->
            // Update the cached copy of the words in the adapter.
            notes?.let { adapter.submitList(it) }
        })
        val fab = findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener {
            val intent = Intent(this@MainActivity, NewNoteActivity::class.java)
            startActivityForResult(intent, newNoteActivityRequestCode)
        }

      

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (requestCode == newNoteActivityRequestCode && resultCode == Activity.RESULT_OK) {
            data?.getStringExtra(NewNoteActivity.EXTRA_REPLY)?.let {
                val note = Note(it)
                noteViewModel.insert(note)
            }
        } else {
            Toast.makeText(
                applicationContext,
                R.string.empty_not_saved,
                Toast.LENGTH_LONG
            ).show()
        }
    }

    override fun onItemClicked(note: TextView) {
        val intent = Intent(this, ModifyNoteActivity::class.java)
        val text = note.text
        
        intent.putExtra("EXTRA_NOTE",text)
        startActivity(intent)
    }

    }

